pkg/nodeps/ddev_environment.go

```diff
@@ -0,0 +1,28 @@
+package nodeps
+
+import (
+	"runtime"
+)
+
+const (
+	DDEVEnvironmentDarwin     = "darwin"
+	DDEVEnvironmentWindows    = "windows"
+	DDEVEnvironmentLinux      = "linux"
+	DDEVEnvironmentWSL2       = "wsl2"
+	DDEVEnvironmentGitpod     = "gitpod"
+	DDEVEnvironmentCodespaces = "codespaces"
+)
+
+// GetDDEVEnvironment returns the type of environment DDEV is being used in
+func GetDDEVEnvironment() string {
+	e := runtime.GOOS
+	switch {
+	case IsCodespaces():
+		e = DDEVEnvironmentCodespaces
+	case IsGitpod():
+		e = DDEVEnvironmentGitpod
+	case IsWSL2():
+		e = DDEVEnvironmentWSL2
+	}
+	return e
+}
```

Comment:
```suggestion
func GetDDEVEnvironment() string {
	switch {
	case IsCodespaces():
		return DDEVEnvironmentCodespaces
	case IsGitpod():
		return DDEVEnvironmentGitpod
	case IsWSL2():
		return DDEVEnvironmentWSL2
	}

	return runtime.GOOS
}
```

Reply:
I like the original better, following the old rule of trying to have one return from a function. This could also be done with a `default: `e = runtime.GOOS`

Reply:
Today early returns is the thing. But anyway does not matter much in this case. The impact here are the superfluous memory allocation you do with the return value, 3 allocations for the same value where one would be fine. This could be really bad for such a helper function which could be called multiple times during runtime.